### PR TITLE
Pull request from CY-8 by Jean Carlo Aguirre

### PR DESCRIPTION
**Pull request by jcaf31**
Jennifer could you please review this Pull Request?

### CODE REVIEW

File: cypress/e2e/cypress_soucedemo_project/inventory.cy.js [hunk 1]
```diff
@@ -18,11 +19,11 @@ describe('Inventory', () => {
     cy.get('[data-test="login-button"]').click();
     cy.url().should('eq', 'https://www.saucedemo.com/inventory.html');
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/inventory.cy.js]: The first part of the URL should go in Fixture file "https://www.saucedemo.com/"

File: cypress/e2e/cypress_soucedemo_project/inventory.cy.js [hunk 2]
```diff
@@ -18,11 +19,11 @@ describe('Inventory', () => {
...
     cy.get('.inventory_item').then(($numberOfItems) => {
-      countItems = $numberOfItems.length;
+      const countItems = $numberOfItems.length;
       for(let i=0; i<countItems; i++){
         cy.get('.inventory_item_img a img').eq(i).invoke('attr', 'src').should('match', new RegExp('.jpg$'));
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/inventory.cy.js]: Please assign it -> new RegExp('.jpg$') in a variable at the top of the it block

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('Shopping Cart', () => {
+
+  let fixture;
+
+  beforeEach(() => {
+    cy.fixture('soucedemo.json').then((fixture_from_fixtures) => {
+      fixture = fixture_from_fixtures;
+    });
+    cy.visit('http://www.saucedemo.com/')
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: the "http" should include "s"?

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  })
+  it('Shopping cart - validate the number of items selected on products page is equal to the cart page', () => {
+
+    let itemsBadgeSelected;
+    let numberOfItemsToAdd;
+    cy.get("#user-name").type(fixture.user_name)
+    cy.get('[data-test="password"]').type(fixture.password)
+    cy.get('[data-test="login-button"]').click()
+    cy.url().should('eq', 'https://www.saucedemo.com/inventory.html')
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: Use the variable in fixtures

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    cy.get('.app_logo').contains('Swag Labs')
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: It "Swag Labs" should be located in Fixtures

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    cy.get('.title').contains('Products')
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: "Products" in fixtures

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      numberOfItemsToAdd = getRandomInt(countItems) + 1;
+      for (let i = 0; i<numberOfItemsToAdd; i++){
+        cy.get('.btn').eq(i).click();
+      }
+    })
+    cy.get('.shopping_cart_badge').then(($badgeValue) => {
+      itemsBadgeSelected = $badgeValue[0].innerText;
+      expect(Number(itemsBadgeSelected)).to.equal(numberOfItemsToAdd, 'The number of items selected were not equal to the number of items displayed in cart badge');
+    })
+    cy.get('#shopping_cart_container').click();
```
Comment (jennicastrop) [inline cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js:36]: After this click, please validate an element of the page is displayed to be sure you are in the correct page

**Review by jennicastrop — CHANGES_REQUESTED** [review]

File: cypress/e2e/cypress_soucedemo_project/inventory.cy.js (hunk 1 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/inventory.cy.js]: The url was added into fixture file as requested and implemented in the 3 test cases

File: cypress/e2e/cypress_soucedemo_project/inventory.cy.js (hunk 2 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/inventory.cy.js:24]: @jennicastrop the porpose of this line is show how it is able to be implemented in 2 differents ways, tell me if is necessary to change?

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js (hunk 3 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: The s was added in the link

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js (hunk 4 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: Work done 👍

File: cypress/e2e/cypress_soucedemo_project/inventory.cy.js (hunk 2 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/inventory.cy.js]: Done

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js (hunk 5 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: Done

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js (hunk 6 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js]: Done

File: cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js (hunk 7 above)
Comment (jcaf31) [reply cypress/e2e/cypress_soucedemo_project/shopping_cart.cy.js:36]: Done

**Comment by jennicastrop** [comment]
Thanks for applying the feedback

**Review by jennicastrop — APPROVED** [review]

**Pull request merged**